python/caffe/pycaffe.py

```diff
@@ -109,8 +116,12 @@ def _Net_forward(self, blobs=None, start=None, end=None, **kwargs):
         outputs = set(self.outputs + blobs)
 
     if kwargs:
-        if set(kwargs.keys()) != set(self.inputs):
-            raise Exception('Input blob arguments do not match net inputs.')
+        expected_kwargs = set(kwargs.keys())
+        received_kwargs = set(self.inputs)
```

Comment:
Expected and received are swapped, or that's my interpretation. Inputs are the expected args (if any).